KachnaOnline.Business/Facades/EventsFacade.cs

```diff
@@ -157,12 +189,54 @@ public async Task ModifyEvent(int eventId, BaseEventDto baseEvent)
         /// Remove an event with the given ID.
         /// </summary>
         /// <param name="eventId">ID of the event to remove.</param>
+        /// <returns>A collection of <see cref="StateDto"/> of states linked to the event at the time of deletion.</returns>
         /// <exception cref="EventNotFoundException">When the event with the given <paramref name="eventId"/> does not
         /// exist.</exception>
         /// <exception cref="EventManipulationFailedException">When the event cannot be deleted.</exception>
-        public async Task RemoveEvent(int eventId)
+        public async Task<IEnumerable<StateDto>> RemoveEvent(int eventId)
+        {
+            var linkedStates = await _eventsService.RemoveEvent(eventId);
+
+            //return linkedStates is { Count: >0 } ? _mapper.Map<List<StateDto>>(linkedStates) : new List<StateDto>() ;
+            return _mapper.Map<List<StateDto>>(linkedStates);
```

Comment:
So do you really want to return nulls if `linkedStates` is null? Why not use `this.MapPlannedStates()`?

Reply:
That is a left-over from the time before `this.MapPlannedStates`. Fixing.